[filter-build-webkit] Show target names in colored output

emw-apple opened the pull request:
#### 3cf2ce928b6ae351943f7da1e8f899ec702ef100
<pre>
[filter-build-webkit] Show target names in colored output
<a href="https://bugs.webkit.org/show_bug.cgi?id=256052">https://bugs.webkit.org/show_bug.cgi?id=256052</a>

Reviewed by Alexey Proskuryakov.

It&apos;s helpful to see which target each line from filter-build-webkit
corresponds with, to see at a glance how far the build has
progressed. When filter-build-webkit is connected to a terminal, prefix
the target name using ANSI colors onto each output line.

The target string is colored based on a checksum of the target name and
project name. This makes the output lines visually distinguishable
(e.g., JavaScriptCore is always the same color), and helps keep the task
description readable.

We only use ANSI &quot;basic&quot; colors (the first 16, excluding black and
white). These colors are set by the user&apos;s terminal theme, so it&apos;s
unlikely that the script will pick a color that is unreadable.

When output is not colorized (i.e. when being read by scripts or run in
CI), the target name is omitted, to preserve readability of the existing
task description.

* Tools/Scripts/filter-build-webkit:
(main):
(printLine):

Canonical link: <a href="https://commits.webkit.org/263507@main">https://commits.webkit.org/263507@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/502891af936b0ecf6c2df1ccc307a7811125b4d5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4828 "Failed to checkout and rebase branch from PR 13049") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4949 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5120 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/6342 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/4955 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [❌ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4820 "Failed to checkout and rebase branch from PR 13049") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5116 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4921 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/6342 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4909 "Failed to checkout and rebase branch from PR 13049") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/4993 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/4326 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/6358 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/2481 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/4321 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/6358 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/4018 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/4343 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/4396 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/6358 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/4566 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4805 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/3915 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/4925 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4312 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/1204 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8375 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/5058 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/552 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4674 "Built successfully") | | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/3/builds/1317 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (approved):
r=me, NICE!!

aproskuryakov commented on Source/JavaScriptCore/unexported-libc++.txt:
```diff
@@ -0,0 +1,34 @@
+# -fvisibilty=hidden, the std:: symbols are exported. This unexport list uses the Itanium mangling rules [1]
+# to attempt to hide all such symbols from the JavaScriptCore executable. As these symbols all come from
+# headers, symbols that slip through are weak and will be caught by the "Check For Weak VTables and Externals"
+# build script phase.
```
I think that the explanation here is a little confusing. AFAICT it's exporting these symbols that's the real problem (perhaps especially new/delete), not that a verifier complains.

But this makes me wonder why libc++ forces symbol visibility. Is it necessary for it in some way?

Super excited about this change though.

emw-apple committed:
[filter-build-webkit] Show target names in colored output
https://bugs.webkit.org/show_bug.cgi?id=256052

Reviewed by Alexey Proskuryakov.

It's helpful to see which target each line from filter-build-webkit
corresponds with, to see at a glance how far the build has
progressed. When filter-build-webkit is connected to a terminal, prefix
the target name using ANSI colors onto each output line.

The target string is colored based on a checksum of the target name and
project name. This makes the output lines visually distinguishable
(e.g., JavaScriptCore is always the same color), and helps keep the task
description readable.

We only use ANSI "basic" colors (the first 16, excluding black and
white). These colors are set by the user's terminal theme, so it's
unlikely that the script will pick a color that is unreadable.

When output is not colorized (i.e. when being read by scripts or run in
CI), the target name is omitted, to preserve readability of the existing
task description.

* Tools/Scripts/filter-build-webkit:
(main):
(printLine):

Canonical link: https://commits.webkit.org/263507@main
